File: Data Structures & Algorithms/majority-element-ii/submission-4.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        int t = n / 3;
        List<Integer> res = new ArrayList<>();
        Map<Integer, Integer> map = new HashMap<>();
        for(int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for(Map.Entry<Integer, Integer> it : map.entrySet()) {
            if(it.getValue() > t) {
                res.add(it.getKey());
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-4.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n, 0);
        for(int i = 0; i < temperatures.size(); i++) {
            for(int j = i + 1; j < temperatures.size(); j++) {
                if(temperatures[j] > temperatures[i]) {
                    res[i] = j - i;
                    break;
                }
            }
        }
        return res;
    }
};
